src/proxy.py

```diff
@@ -50,30 +55,58 @@ def __init__(self, loop):
 
         logging.info('HTTP REQUEST ' + str(request))
 
-        # Check if the request is on the blacklist or whitelist
-        if self.whitelist.check(request):
-            logging.info('Request is on the whitelist')
-        elif self.blacklist.check(request):
-            logging.info('Request is on the blacklist')
-            writer.write(b'HTTP/1.1 404 Not Found\n\n')
+        # this is where we should reject requests that come from unauthorized ip's
+        request_allowed_acl = False
+        try:
+            ip_address = writer.get_extra_info('peername')[0]
+
+            if ip_address is None:
+                logging.info('Could not get remote IP address')
+                writer.write(b'HTTP/1.1 403 Forbidden\n\n')
+                writer.close()
+
+            else:
+                if not self.acl.ip_allowed(ip_address):
+                    logging.info('Request from {} comes from disallowed network per ACL\'s, returning \'HTTP/1.1 403 Forbidden\''.format(ip_address))
+                    writer.write(b'HTTP/1.1 403 Forbidden\n\n')
+                    writer.close()
+
+                else:
+                    logging.info('Request from {} comes from allowed network per ACL\'s, continuing to process request'.format(ip_address))
+                    request_allowed_acl = True
```

Comment:
Drop this variable. And let the allowed requests fall through to the actual request handling code.